velox/docs/functions/presto/array.rst

```diff
@@ -93,21 +93,26 @@ Array Functions
 
 .. function:: array_max(array(E)) -> E
 
-    Returns the maximum value of input array. ::
+    Returns maximum non-NULL element of the array. Returns NULL if array is empty or contains a NULL element.
+    When E is DOUBLE or REAL, returns NaN if any element is NaN. ::
 
         SELECT array_max(ARRAY [1, 2, 3]); -- 3
         SELECT array_max(ARRAY [-1, -2, -2]); -- -1
         SELECT array_max(ARRAY [-1, -2, NULL]); -- NULL
         SELECT array_max(ARRAY []); -- NULL
+        SELECT array_max(ARRAY [-0.0001, -0.0002, -0.0003, nan()]); -- NaN
 
 .. function:: array_min(array(E)) -> E
 
-    Returns the minimum value of input array. ::
+    Returns minimum non-NULL element of the array. Returns NULL if array is empty or contains a NULL element.
+    When E is DOUBLE or REAL, NaN value is considered greater than any non-NaN value. ::
```

Comment:
>Returns NULL if array is empty or contains a NULL element.
> NaN value is considered greater than any non-NaN value.

This doesn't seem to hold true completely.
NaN seems to always be returned in both min_by and max_by even if NULL exists which also means its simultaneously considered both highest and lowest value.
Tried these on presto java:
```
SELECT array_max(ARRAY [nan(), 1.0]); => NaN 
SELECT array_max(ARRAY [NULL, nan(), 1.0]); => NaN 

SELECT array_min(ARRAY [nan(), 1.0]); => NaN 
SELECT array_min(ARRAY [NULL, nan(), 1.0]); => NaN 
```

Reply:
Good catch @bikramSingh91. In Presto, in presence of a NAN, the result is always NAN regardless of the other elements in the array for both min/max functions.